extensions/amp-story/0.1/amp-story-hint.js

```diff
@@ -178,12 +178,14 @@ export class AmpStoryHint {
   }
 
   /**
-   * Shows the given hint, only if not desktop.
+   * Shows the given hint.
    * @param {string} hintClass
    * @private
    */
   showHint_(hintClass) {
-    if (this.storeService_.get(StateProperty.DESKTOP_STATE)) {
+    // Don't show the swiping hint on desktop, or when the share menu is on.
```

Comment:
This is for all hints; I think you want to move this to `showNavigationOverlay`.

As a side note: I was wondering what the best practice is for cases like this.  There's already the state `CAN_SHOW_NAVIGATION_OVERLAY_HINT`; does it make sense for us to, in the store service, subscribe to `DESKTOP_STATE` and `SHARE_MENU_STATE` and set `CAN_SHOW_NAVIGATION_OVERLAY_HINT`?